scripts.Rmd

```diff
@@ -0,0 +1,258 @@
+# `drake` and script-based workflows {#scripts}
+
+```{r, message = FALSE, warning = FALSE, echo = FALSE}
+knitr::opts_knit$set(root.dir = fs::dir_create(tempfile()))
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
+```
+
+```{r, message = FALSE, warning = FALSE, echo = FALSE}
+library(drake)
+library(glue)
+library(purrr)
+library(rlang)
+library(tidyverse)
+invisible(drake_example("script-based-workflows", overwrite = TRUE))
+tmp <- file.copy(
+  list.files("script-based-workflows/R/",
+             pattern = "*.R", full.names = TRUE),
+  ".",
+  recursive = TRUE)
+tmp <- file.copy("script-based-workflows/raw_data.xlsx", ".")
+tmp <- file.copy("script-based-workflows/report.Rmd", ".")
+dir.create("data")
+```
+## Function-oriented workflows
+
+`drake` works best when you write functions for data analysis. Functions break down complicated ideas into manageable pieces.
+
+```{r}
+# R/functions.R
+get_data <- function(file){
+  readxl::read_excel(file)
+}
+
+munge_data <- function(raw_data){
+  raw_data %>% 
+    mutate(Species = forcats::fct_inorder(Species))
+}
+
+fit_model <- function(munged_data){
+  lm(Sepal.Width ~ Petal.Width + Species, munged_data)
+}
+```
+
+By turning the steps of processing into functions like `get_data()`, `munge_data()`, 
+and `fit_model()`, they become special shorthand that make the plan easier to
+read and understand exactly what is being done.
+
+
+```{r}
+# R/plan.R
+plan <- drake_plan(
+  raw_data = get_data(file_in("raw_data.xlsx")),
+  munged_data = munge_data(raw_data),
+  model = fit_model(munged_data)
+)
+```
+
+This function-oriented approach is elegant, powerful, testable, scalable, and 
+maintainable. However, it can be challenging to convert pre-existing traditional
+script-based analyses to function-oriented `drake`-powered workflows. This
+chapter describes a stopgap to retrofit drake and functions onto existing 
+projects. It is quick and painless, and you do not need to change your original 
+scripts.
+
+## Traditional and Legacy Workflows
```

Comment:
Can we use lowercase for "Legacy" and "Workflows"? In the rest of the manual, only the first letter tends to be uppercase except for proper nouns.